### ISSUE TITLE
Feature request: Boot jumper pins breakout

### ISSUE THREAD

**Issue by solarmon** [yet-another-average-joe/Kicad-STM32#2]
Hi,

I'm looking to make use of these excellent Blue Pill symbols and footprints in a project.

I see that you have a symbol and footprint for a Blue Pill with the SWD pins broken out, which I think is very useful and it means I can map them to somewhere else on the board.

Could the same be done for the Boot jumper pins - so that they can also be mapped somewhere else on the board?

**Comment by yet-another-average-joe** [comment]
Unfortunately, the boot pins are not on a known grid. The distance between the boot pin header and pins RESET/B11/B12 is somewhere between 3.81mm and 5.08mm. Could be 4.5mm. Even a 0.1mm tolerance would make the module placement difficult or impossible with 1.00mm holes and 0.6mm pins. This is the reason why there's no such footprint.

**Comment by solarmon** [comment]
Hi.

I'm not sure what you mean by "the boot pins are not on a known grid". I was just expecting to be able to connect the BOOT0 and BOOT1 holes/pins to the board.

It is annoying that the boot pin holes don't line up with the other pins in a standard manner. I see it seems to be lined up with B11, B10 and B1 but it is slightly offset.

Are you saying that different Blue Pills could have different placements for these boot pins? I.e. they are not all a standard design?

**Comment by yet-another-average-joe** [comment]
There are only two ways to konw where this pin header exactly seats : measure better than 0.05mm, or estimate, then order some PCBs and see if the header fits.
Measuring accurately is not easy. The way to go should be from a naked BluePill PCB (without the header), using an optical comparator. (or using the poor man's comparator : a microscope or macro photo lens on a DSLR + photo metrology software - not sure for the termilogy !).

See below : what values for A and B ???

![YAAJ_BluePill_1](https://user-images.githubusercontent.com/43514870/100881023-bd02dc80-34ad-11eb-9ec9-41c970418c5c.png)

**Comment by solarmon** [comment]
Hi,

I've had a go myself. I've removed the jumpers and I have digital calipers to try to do some measurements.

![image](https://user-images.githubusercontent.com/46369787/100907372-27257c80-34c2-11eb-8456-4717221c1fe4.png)

However, the measurements I've made are just by eye/hand with the digital calipers at the moment and will not be completely accurate, but the picture gives an indication of where the holes are.

But like you said, only getting it printed and tested will only prove whether it is correct or not.

**Comment by yet-another-average-joe** [comment]
Yes, calipers + eye balling will not give accurate enough measurements. But a good macro photo, shot with a tablet displaying a white screen as background, knowing the board dimensions +  appropriate software should give good results (never did it on my side, so no idea for the software).

Your picture definitely shows that there are different BluePill layouts for the Boot pinheader : mine are more or less facing 37-36-35. (just verified two batches of 10, I got 2 years and 1 year ago). Now we probably get clones of clones from what was already clones of clones of clones ! (not sure they shared the Gerber files all over China !)

Therefore, only the footprints with SWD and the main 40 pins could be shared.
